api/src/services/streams-service.ts

```diff
@@ -84,42 +85,44 @@ export class StreamsService {
 
 	getLogs = async (
 		subscription: Author | Subscriber
-	): Promise<{ publicData: string[]; maskedData: string[]; subscription: Author | Subscriber; latestLink: string }> => {
-		try {
-			let exists = true;
-			let publicData: string[] = [];
-			let maskedData: string[] = [];
-			let latestLink = '';
-
-			while (exists) {
-				let next_msgs: any = [];
-
-				next_msgs = await subscription.clone().fetch_next_msgs();
-
-				if (next_msgs.length === 0) {
-					exists = false;
-				} else {
-					latestLink = next_msgs[next_msgs.length - 1]?.get_link()?.to_string();
-				}
-
-				for (let i = 0; i < next_msgs.length; i++) {
-					const pubPayload = next_msgs[i].get_message().get_public_payload();
-					const maskedPayload = next_msgs[i].get_message().get_masked_payload();
-					publicData = [...publicData, fromBytes(pubPayload)];
-					maskedData = [...maskedData, fromBytes(maskedPayload)];
-				}
+	): Promise<{ channelData: ChannelData[]; subscription: Author | Subscriber; latestLink: string }> => {
+		let foundNewMessage = true;
+		let channelData: ChannelData[] = [];
+		let latestLink = '';
+
+		while (foundNewMessage) {
+			let next_msgs: any = [];
+
+			next_msgs = await subscription.clone().fetch_next_msgs();
+
+			if (next_msgs.length === 0) {
+				foundNewMessage = false;
+			} else {
+				latestLink = next_msgs[next_msgs.length - 1]?.get_link()?.to_string();
 			}
 
-			return {
-				publicData,
-				maskedData,
-				subscription,
-				latestLink
-			};
-		} catch (error) {
-			console.log('Error from streams sdk:', error);
-			throw new Error('could not get logs from the channel');
+			if (next_msgs && next_msgs.length > 0) {
+				const cData = next_msgs.map((userResponse: any) => {
```

Comment:
might be helpful to define a type of the expected `userResponse` object

Reply:
the streams lib for now does not provide a proper data model. I'll need to wait here for datatypes of them

Reply:
understand, however it is hard to believe that the Streams team will have time for this in the next several weeks. 
If you already have an object which you can inspect, would it be possible to create a type using the current values?

Reply:
Ok, for now I don't know about their detailed data models but I will ask them or look into the spec of Streams for future PRs for now I would keep it as it is